### DISCUSSION TITLE
How to shift the phase of a lightcurve

### DISCUSSION THREAD

**Discussion by kh-kr** [phoebe-project/phoebe2#582] [General]
Hi,

![Screenshot from 2022-04-03 00-25-17](https://user-images.githubusercontent.com/79892737/161397434-9a2929b5-7e3f-47de-a61f-863a9b0ba1d4.png)

I am trying to fit a TESS lightcurve to get parameters of primary and secondary components of an eclipsing binary. I am able to fit it but not able to shift the phase as the smaller dip is not visible in the -0.4 to 0.4 phase range.

And also the blue curve isn't fitting to the lightcurve.
Can anyone help me regarding this?

Thanks!
Khushboo

**Comment by kecnry** [comment]
Hi Khushboo,

Any phase or time-shift of the model itself relative to the data is handled by the [various t0 parameters in phoebe](http://phoebe-project.org/docs/latest/tutorials/t0s).  If you just want to shift the phase-range that you're plotting (plotting from -0.6 to 0.4 instead of -0.5 to 0.5, for example), you'll unfortunately need to handle that plotting manually.

As for the blue curve, we'll need more information about what and how that was computed to help.  I'm guessing it might just be an issue of not providing the fitting algorithm any flexibility to change the overall light levels - those are mostly handled through [passband luminosity parameters and options](http://phoebe-project.org/docs/latest/tutorials/pblum).

Hope this helps!
Kyle

**Comment by kh-kr** [comment]
Hi Kyle,

Thank you! Now I am able to shift the phase as you can see in the below figure.

![Screenshot from 2022-04-05 12-06-57](https://user-images.githubusercontent.com/79892737/161694034-da64fbbb-b3c8-4fc2-a0b9-c8a784b1557e.png)

The model gives me temperatures, logg and various other parameters, but I have few other doubts-

1) How can I get the luminosities and radii of both the components of the system? 
2) How can I remove the latest and default curves from the plot because I need to create a final figure?  

Best regards,
Khushboo

**Reply by kecnry** [reply]
There are input parameters for the radii of the stars (`requiv` parameters) which you can read out (`print(b.filter(qualifier='requiv'))`, for example).  The luminosities are not always input parameters (it depends on what you set for `pblum_mode`).  You can access how those luminosities are interpreted, at time t0, by calling [compute_pblums](http://phoebe-project.org/docs/latest/api/phoebe.frontend.bundle.Bundle.compute_pblums) or if you need them at other times, you can expose them in a [mesh dataset](http://phoebe-project.org/docs/latest/tutorials/meshes).

**Comment by wpr1820** [comment]
Hi，sir. I'm using one cycle of data, but the plotted image seems to have only half a cycle. when I use time as x axis it works fine.

b.add_dataset('lc', times=times, fluxes=fluxes, sigmas=sigmas, dataset='lc01', overwrite=True)
b.run_compute(irrad_method='none')
afig, mplfig = b['lc01@dataset'].plot(x='phases', yerror=None, s=0.01, c='black', show=True)
![Figure_1](https://user-images.githubusercontent.com/112557810/191695620-e3590fcf-b611-46fb-b0aa-f816acf2f3ae.png)
![Figure_1time](https://user-images.githubusercontent.com/112557810/191695684-406277f9-63ed-4e68-85dd-ad02fb7ee47a.png)
Can you help me see where I am going wrong.
Thanks！

**Reply by kecnry** [reply]
My first guess without seeing the parameters is that your `times` array doesn't actually cover a full orbital cycle (although the plot vs time does make it look like it should).  If you're sure that isn't the case, then can you share a full minimal example to reproduce what you're seeing?  It could also be a bug with the phasing or perhaps a parameter-mapping issue if you're using a different backend.

-Kyle

**Reply by wpr1820** [reply]
OK. My object is KIC 11560447, which cycle is 0.5276790. I am using sc data.From the BJD of the data, I'm sure there is only have one cycle. 
Here is a graph I made using the two period data.
![Figure_2](https://user-images.githubusercontent.com/112557810/191752427-d859a2ed-dcbe-4faf-8d9a-2fa5ac59d42f.png)
![Figure_2time](https://user-images.githubusercontent.com/112557810/191753178-6a02d73c-05b0-43c5-85c6-a0396cbd328a.png)
When I don't add x='phase' to the plot, everything works fine

**Reply by kecnry** [reply]
What do you have set as the orbital period in phoebe?  It looks from the time-series data that the orbital period should be around 0.6 days, but phoebe defaults to 1 day (and uses the orbital period for phasing).

**Reply by wpr1820** [reply]
Ohhhh, I forgot to set the period. I have done this problem.
![Figure_good](https://user-images.githubusercontent.com/112557810/191765099-0db1b8e1-d86d-48f4-b09d-684e4cd668b0.png)
…]()
Thank you very much!
